src/base/bittorrent/session.cpp

```diff
@@ -1405,9 +1405,14 @@ bool Session::deleteTorrent(const QString &hash, bool deleteLocalFiles)
 
     // Remove it from session
     if (deleteLocalFiles) {
-        QString rootPath = torrent->rootPath(true);
-        if (!rootPath.isEmpty())
-            m_savePathsToRemove[torrent->hash()] = rootPath;
+        if (torrent->savePath(true) == torrentTempPath(torrent->hash())) {
+            m_savePathsToRemove[torrent->hash()] = torrent->savePath(true);
+        }
+        else {
+            QString rootPath = torrent->rootPath(true);
+            if (!rootPath.isEmpty())
+                m_savePathsToRemove[torrent->hash()] = rootPath;
+        }
         m_nativeSession->remove_torrent(torrent->nativeHandle(), libt::session::delete_files);
     }
     else {
```

Comment:
I thought about it... but I decided to wait for your feedback first. I would have deleted the temporary folder unconditionally, but many users misunderstand its meaning (temp folder) and they will be surprised that their files are deleted when they did not choose this option.

We need to figure this out definitely. Either we explicitly give this responsibility to users, or we will tighten the "temporality" of that folder (so the user clearly understood that it cannot and should not control it and own it).